ambari-server/src/main/java/org/apache/ambari/server/scheduler/ExecutionScheduleManager.java

```diff
@@ -427,8 +444,64 @@ protected String getJobName(Long executionId, Long orderId) {
    */
   public void updateBatchSchedule(RequestExecution requestExecution)
     throws AmbariException {
+    BatchRequest lastCompletedBatchRequest = calculateLastCompletedBatch(requestExecution);
+    //TODO check if last completed was last at all.
+
+    if (requestExecution.getStatus().equals(SCHEDULED.name())) {
+      long startingBatchId = lastCompletedBatchRequest == null? 1L : lastCompletedBatchRequest.getOrderId() + 1;
+      scheduleBatch(requestExecution, startingBatchId);
+    } else if (requestExecution.getStatus().equals(PAUSED.name()) ||
+               requestExecution.getStatus().equals(ABORTED.name())) {
+      //TODO delete all or only pending?
+      //TODO delete only trigger?
+      deleteAllJobs(requestExecution);
+      long possibleBatchToAbort = lastCompletedBatchRequest == null? 1L : lastCompletedBatchRequest.getOrderId() + 1;
+      //TODO abort only in_progress?
+      Collection<Long> requestIDsToAbort = requestExecution.getBatchRequestRequestsIDs(possibleBatchToAbort);
+      //TODO move to the separate method
+      for (Long requestId : requestIDsToAbort) {
+        //might be null if the request is for not long running job
+        if (requestId == null) continue;
+
+        StrBuilder sb = new StrBuilder();
+        sb.append(DEFAULT_API_PATH)
+          .append("/clusters/")
+          .append(requestExecution.getClusterName())
+          .append("/requests/")
+          .append(requestId);
+
+        String body = "{\"Requests\":{\"request_status\":\"ABORTED\",\"abort_reason\":\"Request schedule status changed to " + requestExecution.getStatus() + "\"}}";
+        //TODO race condition : request during sending the abort, what should we do?
+        performApiRequest(sb.toString(), body, "PUT", requestExecution.getAuthenticatedUserId());
```

Comment:
This is flaky, we do not need to use API endpoints to cancel request, just get the ResourceProvider for Requests and use PreportyHelpers to achieve the same. It makes sense for unit testing as well. Also please add unit tests around this.